small-talk/db/registerTest.js

```diff
@@ -0,0 +1,28 @@
+"use server"
+const uri = "mongodb+srv://smt_root:pokemonwithguns@smalltalkcluster0.jo4jne6.mongodb.net/?retryWrites=true&w=majority"
+import Mongoboi from "./mongo"
+export default async function handleRegister(username, password, firstname, lastname, dob){
+  const mongoboi = new Mongoboi(uri, "Users");
+  await mongoboi.connect();
+
+  const newUser = {
+    username: username,
+    password: password,
+    firstname: firstname,
+    lastname: lastname,
+    dob: dob,
+  };
+
+  try {
```

Comment:
I recommend appending hthe finally block to the try catch blocks.

```suggestion
  try {
    await mongoboi.connect();
    await mongoboi.insertOne("patients", newUser);
  // stuff
  } catch (error)
  {
    console.log(error);
  }
  finally {
   await mongoboi.disconnect();
  }
```

You can be sure that the connection is closed because the finally block will run any code despite any exceptions being thrown.

Reply:
We will be sure to address this in a separate issue. Thank you